src/raftstore/store/peer.rs

```diff
@@ -541,6 +550,17 @@ impl Peer {
         if self.is_leader() && m.get_from() != INVALID_ID {
             self.peer_heartbeats.insert(m.get_from(), Instant::now());
         }
+        if let Entry::Occupied(e) = self.peer_pending_after.entry(m.get_from()) {
+            let status = self.raft_group.status();
+            let truncated_idx = self.raft_group.get_store().truncated_index();
+            if let Some(progress) = status.progress.get(&m.get_from()) {
```

Comment:
What if it's a learner?

Reply:
How about consider it later? Because for now we have not insert the peer into `peer_pending_after` after `AddLearnerNode` finished.